lms/static/js/ccx/schedule.js

```diff
@@ -104,23 +104,37 @@ var edx = edx || {};
 	    // Add unit handler
 	    $('#add-unit-button').on('click', function(event) {
 	      event.preventDefault();
+	      // Default value of time is 00:00.
+	      var start = self.get_datetime('start');
 	      var chapter = self.chapter_select.val(),
-		  sequential = self.sequential_select.val(),
-		  vertical = self.vertical_select.val(),
-		  units = self.find_lineage(self.schedule,
-		    chapter,
-		    sequential === 'all' ? null : sequential,
-		    vertical === 'all' ? null: vertical),
-		  start = self.get_datetime('start'),
-		  due = self.get_datetime('due');
-	      units.map(self.show);
-	      var unit = units[units.length - 1];
-	      if (unit !== undefined && start) { unit.start = start; }
-	      if (unit !== undefined && due) { unit.due = due; }
-	      self.schedule_apply([unit], self.show);
-	      self.schedule_collection.set(self.schedule);
-	      self.dirty = true;
-	      self.render();
+	      sequential = self.sequential_select.val(),
+	      vertical = self.vertical_select.val(),
+	      units = self.find_lineage(
+	        self.schedule,
+	        chapter,
+	        sequential === 'all' ? null : sequential,
+	        vertical === 'all' ? null : vertical
+	      ),
+	      due = self.get_datetime('due');
```

Comment:
Why are you removing the indentation here? It makes it more difficult to see that these lines are all part of the same statement. If you want, you can initialize the local variables all on one line, and then assign to them without indentation -- that would be clearer to me. For example:
```js
var start, chapter, sequential, vertical, units, due;
start = self.get_datetime('start');
sequential = self.sequential_select.val();
// etc...
```